src/bot_cog.py

```diff
@@ -178,7 +178,20 @@ async def _set_carpool_logic_admin(
             meta_data.carpool_calculation_method = CarpoolCalculationMethod.VARSITY_IN_NOVICE
         meta_data.save()
         await ctx.send(f'Successfully updated the carpool logic.')
-    
+
+    #New Command Written by robert-----------------------------------------------------------------------------------------------------
+    @commands.command(name="extract-carpool-data-admin", brief='Releases a list of all historical carpools')
+    async def _extract_carpool_data_admin(
+        self,
+        ctx,
+        #My understanding is that ctx is somehow checking the role of the user, as well as the channel is that correct?
```

Comment:
Well, `ctx` gives us the data that allows us to validate the role of the user and the channel when we call `admin_validation(ctx)` and `channel_validation(ctx)`.